docker/Dockerfile

```diff
@@ -0,0 +1,107 @@
+    python2 \
+    python3-matplotlib \
+    unzip \
+    wget
+
+RUN pip install --upgrade pip==21.2.3
+RUN pip install pip-tools==6.6.2 setuptools==58.5.3 cmake
+
+# Installing google cloud sdk
+RUN wget -O gcloud-sdk.tar.gz https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-364.0.0-linux-x86_64.tar.gz && \
+    tar -xzf gcloud-sdk.tar.gz && \
+    ./google-cloud-sdk/install.sh --quiet && \
+    rm gcloud-sdk.tar.gz
+
+WORKDIR /app/oppia
+
+# Installing python dependencies from the requirements_dev.txt file
+COPY requirements_dev.txt .
+
+RUN pip install -r requirements_dev.txt
+
+# Installing buf and proto for Linux -- this docker container is based on Linux
+COPY buf.gen.yaml .
+COPY /extensions/classifiers/proto/ ./extensions/classifiers/proto/
+
+# Installing third party dependencies
+COPY scripts ./scripts
+COPY /core/utils.py ./core/utils.py
+COPY /core/feconf.py ./core/feconf.py
+COPY /core/constants.py ./core/constants.py
+COPY /assets/constants.ts ./assets/constants.ts
+COPY /assets/release_constants.json ./assets/release_constants.json
+COPY dependencies.json .
+
+# Installation for proto buf requires the third party dependencies to be installed (oppia-ml-proto)
+# This also downloads frontend dependencies, which should be removed once the dependencies.json is deprecated
+RUN python -m scripts.install_dependencies_json_packages
+RUN python -m scripts.build
+
+ENV BUF_LINUX_FILES="buf-Linux-x86_64 protoc-gen-buf-check-lint-Linux-x86_64 protoc-gen-buf-check-breaking-Linux-x86_64" \
+    PROTOC_LINUX_FILE='protoc-3.13.0-linux-x86_64.zip' \
+    BUF_DIR='/app/buf-0.29.0' \
+    PROTOC_DIR='/app/buf-0.29.0/protoc' \
+    BUF_BASE_URL='https://github.com/bufbuild/buf/releases/download/v0.29.0/' \
+    PROTOC_URL='https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0'
+
+# Download BUF_LINUX_FILES and PROTOC_LINUX_FILE
+# Set the shell as /bin/bash explicitly. This is required to make use of the array syntax correctly
+SHELL ["/bin/bash", "-c"]
+RUN IFS=' ' read -ra BUF_FILES <<< "$BUF_LINUX_FILES"; \
+    for bin_file in "${BUF_FILES[@]}"; do \
+        wget -P $BUF_DIR $BUF_BASE_URL/$bin_file; \
+    done
+
+RUN wget -P $BUF_DIR $PROTOC_URL/$PROTOC_LINUX_FILE
+RUN unzip $BUF_DIR/$PROTOC_LINUX_FILE -d $PROTOC_DIR \
+    && rm $BUF_DIR/$PROTOC_LINUX_FILE
+
+RUN chmod -R 744 $BUF_DIR \
+    && chmod -R 744 $PROTOC_DIR
+
+# Compiling the protobuf files
+ENV PROTOC_FILES_PATH='/app/oppia/third_party/oppia-ml-proto-0.0.0'
+
+RUN npm install protoc-gen-ts@0.3.4 google-protobuf@3.13.0 typescript@4.1.6
+RUN mkdir /app/oppia/proto_files
+RUN /app/buf-0.29.0/protoc/bin/protoc \
+    --plugin=protoc-gen-ts=/app/oppia/node_modules/.bin/protoc-gen-ts \
+    --ts_out=/app/oppia/extensions/classifiers/proto/ \
+    --js_out=import_style=commonjs,binary:/app/oppia/extensions/classifiers/proto/ \
+    --python_out=/app/oppia/proto_files \
+    --proto_path=$PROTOC_FILES_PATH \
+    $PROTOC_FILES_PATH/*.proto
```

Comment:
Can we move proto-buf related things before pip install?

Reply:
If not, let's move wget and unzip part on the top so that we don't have to rerun it if requirements.txt changes

Reply:
done